launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/PollingDataSource.java

```diff
@@ -87,7 +94,19 @@ public void stop(Callback<Void> completionCallback) {
     }
 
     private void poll(Callback<Boolean> resultCallback) {
-        ConnectivityManager.fetchAndSetData(fetcher, context, dataSourceUpdateSink,
-                resultCallback, logger);
+        // poll if there is no max (negative number) or there are polls remaining
+        if (maxNumberOfPolls < 0 || numberOfPollsRemaining > 0) {
+            ConnectivityManager.fetchAndSetData(fetcher, context, dataSourceUpdateSink,
+                    resultCallback, logger);
+            numberOfPollsRemaining--; // decrementing even when we have unlimited polls has no consequence
```

Comment:
For reviewers: Decrementing with unlimited polls will underflow after 20428 years.  Seemed worth the simplified implementation.